web/packages/core/src/ruffle-player.ts

```diff
@@ -816,6 +822,21 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Helper function to get the distance between two points
+     *
+     * @param x1 The x coordinate of point 1.
+     * @param y1 The y coordinate of point 1.
+     * @param x2 The x coordinate of point 2.
+     * @param y2 The y coordinate of point 2.
+     * @returns The distance between the two points.
+     */
+    getDistance(x1: number, y1: number, x2: number, y2: number): number {
```

Comment:
The logic looks good in general, thank you!

One nitpick - I think the more TypeScripty way to handle points would be to introduce:
```ts
class Point {
    constructor(public x: number, public y: number) {}

    distanceTo(other: Point) {
        const dx = other.x - this.x;
        const dy = other.y - this.y;
        return Math.sqrt(dx * dx + dy * dy);
    }
}
```
Then use it in the various places. What do you think?

Reply:
I agree that that's cleaner. Added, thanks!